nncf/experimental/tensor/functions/torch_numeric.py

```diff
@@ -191,6 +192,28 @@ def _(a: torch.Tensor, decimals=0) -> torch.Tensor:
     return torch.round(a, decimals=decimals)
 
 
+@numeric.quantile.register(torch.Tensor)
+def _(
+    a: torch.Tensor,
+    q: Union[float, List[float]],
+    axis: Optional[Union[int, Tuple[int]]] = None,
+    keepdims: Optional[bool] = None,
+) -> torch.Tensor:
+    device = a.device
+    # See https://github.com/pytorch/pytorch/issues/61582
+    # https://github.com/pytorch/pytorch/issues/64947
+    if len(a) <= 16_000_000 and isinstance(axis, int):
+        result = torch.quantile(
+            a,
+            torch.tensor(q, dtype=a.dtype, device=a.device),
+            axis,
+            keepdims,
+        )
+    else:
+        result = torch.tensor(np.quantile(a.detach().cpu().numpy(), q=q, axis=axis, keepdims=keepdims))
+    return result.type(a.dtype).to(device)
```

Comment:
In numpy quantile always return float64. But in torch it can be float32 or float64 depends from input dtype. 
In your implementations it will cast to input dtype.
Think that need to return dtype like in numpy. 

And torch implementation supports only float32 and float64, in other case need to use numpy.

```python 
if len(a) <= 16_000_000 and isinstance(axis, int) and a.dtype in [torch.float32, torch.float64]:
    return torch.quantile(...).to(torch.float64)
return torch.tensor(np.quantile(...)).to(device)
```

Reply:
I hope it won't broke quantization parameters precision 🙏

Reply:
Done